datetime/src/main/java/com/fasterxml/jackson/datatype/jsr310/deser/DurationDeserializer.java

```diff
@@ -79,20 +88,38 @@ protected DurationDeserializer withLeniency(Boolean leniency) {
                     deser = deser.withLeniency(leniency);
                 }
             }
+            if (format.hasPattern()) {
+                this.pattern = DurationPattern.from(format.getPattern())
+                        .orElse(null);
```

Comment:
Wondering if this should follow the "pattern" of other configuration settings, and set pattern in the super class, and then call a method like the following:
`deser = deser.withPattern(format.getPattern());`

This would set the groundwork for other types adding support for 'pattern's, which might help with some tricky bugs. Or is this duplicating work in date-time-config? Thoughts @cowtowncoder ?

Reply:
That could make sense, but I think is a bigger topic so I Ieft the disussion up to you :)

Reply:
But in any case, adding a unit in JsonFormat is not a conflict with the current change and it can be added later on.